File: socialgo/src/main/java/com/fungo/socialgo/listener/impl/SimpleShareListener.java
package com.fungo.socialgo.listener.impl;

import com.fungo.socialgo.exception.SocialError;
import com.fungo.socialgo.listener.OnShareListener;
import com.fungo.socialgo.model.ShareObj;

/**
 * CreateAt : 2017/5/22
 * Describe : 简化版本分享监听
 *
 * @author chendong
 */
public class SimpleShareListener implements OnShareListener {

    @Override
    public void onStart(int shareTarget, ShareObj obj) {
    }

    @Override
    public ShareObj onPrepareInBackground(int shareTarget, ShareObj obj) throws Exception {
        return obj;
    }

    @Override
    public void onSuccess() {

    }

    @Override
    public void onFailure(SocialError e) {

    }

    @Override
    public void onCancel() {

    }
}

File: socialgo/src/main/java/com/fungo/socialgo/listener/impl/SimpleShareListener.kt
package com.fungo.socialgo.listener.impl

import com.fungo.socialgo.exception.SocialError
import com.fungo.socialgo.listener.OnShareListener
import com.fungo.socialgo.model.ShareEntity

/**
 * 分享回调简单实现
 */
class SimpleShareListener : OnShareListener {

    override fun onStart(shareTarget: Int, obj: ShareEntity) {}

    @Throws(Exception::class)
    override fun onPrepareInBackground(shareTarget: Int, obj: ShareEntity): ShareEntity {
        return obj
    }

    override fun onSuccess() {

    }

    override fun onFailure(e: SocialError) {

    }

    override fun onCancel() {

    }
}
